src/common/providers/identity.ts

```diff
@@ -199,181 +426,62 @@ export interface DecodedJwt {
 }
 
 /**
- * Helper class to create the user metadata in a UserRecord object
- */
-export class UserRecordMetadata implements UserMetadata {
-  constructor(public creationTime: string, public lastSignInTime: string) {}
-
-  /** Returns a plain JavaScript object with the properties of UserRecordMetadata. */
-  toJSON(): object {
-    return {
-      creationTime: this.creationTime,
-      lastSignInTime: this.lastSignInTime,
-    };
-  }
-}
-
-/**
- * Helper class to create the user info in a UserRecord object
- */
-export class UserRecordInfo implements UserInfo {
-  constructor(
-    public uid: string,
-    public displayName: string,
-    public email: string,
-    public photoURL: string,
-    public providerId: string,
-    public phoneNumber: string
-  ) {}
-
-  toJSON(): object {
-    return {
-      uid: this.uid,
-      displayName: this.displayName,
-      email: this.email,
-      photoURL: this.photoURL,
-      providerId: this.providerId,
-      phoneNumber: this.phoneNumber,
-    };
-  }
-}
-
-/**
- * Helper class to create the user MultiFactorInfo in a UserRecord object
- */
-export class UserRecordMultiFactorInfo
-  implements Pick<MultiFactorInfo, keyof MultiFactorInfo> {
-  constructor(
-    public uid: string,
-    public factorId: string,
-    public displayName?: string,
-    public enrollmentTime?: string
-  ) {}
-
-  toJSON(): object {
-    return {
-      uid: this.uid,
-      factorId: this.factorId,
-      displayName: this.displayName || null,
-      enrollmentTime: this.enrollmentTime || null,
-    };
-  }
-}
-
-/**
- * Helper class to create the user PhoneMultiFactorInfo in a UserRecord object
+ * @internal
+ * Helper to determine if we refresh the public keys
  */
-export class UserRecordPhoneMultiFactorInfo
-  implements Pick<PhoneMultiFactorInfo, keyof PhoneMultiFactorInfo> {
-  constructor(
-    public uid: string,
-    public factorId: string,
-    public phoneNumber: string,
-    public displayName?: string,
-    public enrollmentTime?: string
-  ) {}
-
-  toJSON(): object {
-    return {
-      uid: this.uid,
-      factorId: this.factorId,
-      phoneNumber: this.phoneNumber,
-      displayName: this.displayName || null,
-      enrollmentTime: this.enrollmentTime || null,
-    };
+export function invalidPublicKeys(
+  keys: PublicKeysCache,
+  time: number = Date.now()
+): boolean {
+  if (!keys.publicKeysExpireAt) {
+    return true;
   }
+  return time >= keys.publicKeysExpireAt;
```

Comment:
per comments in test, this should be `time + INVALID_TOKEN_BUFFER >= keys.publicKeysExpireAt`

Reply:
fixed